fdbserver/ClusterController.actor.cpp

```diff
@@ -3084,6 +3097,63 @@ class ClusterControllerData {
 	}
 };
 
+// Wrapper for singleton interfaces
+template <class Interface>
+struct Singleton {
+	const Optional<Interface>& interface;
+
+	Singleton(const Optional<Interface>& interface) : interface(interface) {}
+
+	virtual Role getRole() const = 0;
+	virtual ProcessClass::ClusterRole getClusterRole() const = 0;
+
+	virtual void setOnDb(ClusterControllerData* cc) const = 0;
```

Comment:
This name is confusing. I'd propose `setInterfaceToDbInfo`

Reply:
Done